planner/core/rule_inject_extra_projection.go

```diff
@@ -40,18 +41,23 @@ func (pe *projInjector) inject(plan PhysicalPlan) PhysicalPlan {
 	for _, child := range plan.Children() {
 		pe.inject(child)
 	}
+
 	switch p := plan.(type) {
 	case *PhysicalHashAgg:
 		plan = injectProjBelowAgg(plan, p.AggFuncs, p.GroupByItems)
 	case *PhysicalStreamAgg:
 		plan = injectProjBelowAgg(plan, p.AggFuncs, p.GroupByItems)
+	case *PhysicalSort:
```

Comment:
```suggestion
	case *PhysicalSort, *PhysicalTopN:
```

Reply:
It seems that we can not merge these two types into one `case` branch though they hold the same attribute.